### PR TITLE
filter out terminated containers in cadvisor_stats_provider

### LINKED ISSUE #103368: Kubelet should not report metrics from init containers
#### What would you like to be added:
as discussed in issue [#792](https://github.com/kubernetes-sigs/metrics-server/issues/792) and[#796](https://github.com/kubernetes-sigs/metrics-server/issues/796) , we get the metrics of terminated container. metric information about terminated pods is expected? If so,  that has changed recently?
#### Why is this needed:
 Kubelet endpoints and CRI metrics protos should report metrics for terminated container should be standardized

### PR DESCRIPTION
**Pull request by 249043822**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
if the container has only one instance, we should still check whether it was terminated(as cpu and memory 0), since init-container exits and would not be restarted, it may have only one exited instance 
#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/103368
https://github.com/kubernetes-sigs/metrics-server/issues/976

#### Special notes for your reviewer:
/cc @haircommander @serathius @yangjunmyfm192085 
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by yangjunmyfm192085 — COMMENTED** [review]
This problem occurs frequently, and there have been many issues. But I did not reproduce in the docker environment.
Perhaps in a cri-o scene. As long as the exited container cgroup still exists, there should be an exception.
I don't know how this scene was originally designed?
But from the current situation, it is still recommended to deal with it.

File: pkg/kubelet/stats/cadvisor_stats_provider.go [hunk 1]
```diff
@@ -340,12 +340,10 @@ func filterTerminatedContainerInfoAndAssembleByPodCgroupKey(containerInfo map[st
 	}
 	result := make(map[string]cadvisorapiv2.ContainerInfo)
 	for _, refs := range cinfoMap {
-		if len(refs) == 1 {
```
Comment (yangjunmyfm192085) [inline pkg/kubelet/stats/cadvisor_stats_provider.go:343]: Changed the original behavior. Need to consider whether other scenarios are also ok

Comment (249043822) [reply pkg/kubelet/stats/cadvisor_stats_provider.go:343]: If only one container instance, it still may be an exited instance(for init container), so we should not process len() == 1 specially

**Comment by serathius** [comment]
cc @dashpole

**Review by yangjunmyfm192085 — COMMENTED** [review]
/test pull-kubernetes-node-e2e-containerd

**Comment by haircommander** [comment]
/lgtm

this makes sense to me! thanks @249043822

**Pull request merged**